packages/editor/src/components/browser/CmsBrowser.tsx

```diff
@@ -1,15 +1,151 @@
+    id: CMS_BROWSER_ID,
+    name: 'CMS',
+    content: <CmsBrowser value={value} onChange={setValue} />,
+    accept: () => createFullPathCMS(value)
+  };
 };
 
-const CmsBrowser = (props: { value: string; onChange: (value: string) => void }) => {
-  return <Input {...props} />;
+export const createFullPathCMS = (value: string) => {
+  const fullPath = value.length === 0 ? '' : `ivy.cms.co("${value}")`;
+  return fullPath;
+};
+
+const CmsBrowser = ({ value, onChange }: { value: string; onChange: (value: string) => void }) => {
+  const { context } = useEditorContext();
+  const tree = useMeta('meta/cms/tree', { context, requiredProjects: false }, []).data;
+  const [selectedContentObject, setSelectedContentObject] = useState<ContentObject>({
+    name: '',
+    children: [],
+    fullPath: '',
+    type: 'STRING',
+    values: {}
+  });
+  const [showHelper, setShowHelper] = useState<boolean>(false);
+
+  const columns = useMemo<ColumnDef<ContentObject>[]>(
+    () => [
+      {
+        accessorFn: row => `${row.name} (${row.type})`,
+        id: 'name',
+        cell: cell => {
+          if (cell.row.original.type === 'FOLDER') {
+            return <ExpandableCell cell={cell} title={cell.row.original.name} additionalIcon={'GoToSource'} />;
+          } else {
+            return <ExpandableCell cell={cell} title={cell.row.original.name} additionalIcon={''} />;
+          }
+        }
+      }
+    ],
+    []
+  );
+
+  const [expanded, setExpanded] = useState<ExpandedState>(false);
+  const [globalFilter, setGlobalFilter] = useState('');
+  const [rowSelection, setRowSelection] = useState<RowSelectionState>({});
+
+  const table = useReactTable({
+    data: tree,
+    columns: columns,
+    state: {
+      expanded,
+      globalFilter,
+      rowSelection
+    },
+    filterFromLeafRows: true,
+    enableRowSelection: true,
+    enableMultiRowSelection: false,
+    enableSubRowSelection: false,
+    onExpandedChange: setExpanded,
+    onGlobalFilterChange: setGlobalFilter,
+    onRowSelectionChange: setRowSelection,
+    getSubRows: row => row.children,
+    getCoreRowModel: getCoreRowModel(),
+    getExpandedRowModel: getExpandedRowModel(),
+    getFilteredRowModel: getFilteredRowModel()
+  });
+
+  useEffect(() => {
+    if (Object.keys(rowSelection).length !== 1) {
+      setSelectedContentObject({ name: '', children: [], fullPath: '', type: 'STRING', values: {} });
+      setShowHelper(false);
+      onChange('');
+      return;
+    }
+    const selectedRow = table.getRowModel().rowsById[Object.keys(rowSelection)[0]];
+    setSelectedContentObject(selectedRow.original);
+    setShowHelper(true);
+    onChange(selectedRow.original.fullPath);
+  }, [onChange, rowSelection, table]);
+
+  return (
+    <>
+      <Table search={{ value: globalFilter, onChange: setGlobalFilter }}>
+        <tbody>
+          {table.getRowModel().rows.map(row =>
+            row.original.type === 'FOLDER' ? (
+              <MessageRow key={row.id}>
```

Comment:
you don't display a message here right? So why not simply use a normal row? `<tr key={row.id}>`

Reply:
or even better improve the SelectRow by an attribute if the row is selectable or not. then you don't have to do this condition and duplication code here!